packages/commonwealth/server/scripts/setupAppRoutes.ts

```diff
@@ -133,92 +133,96 @@ const setupAppRoutes = (
-          where: { id: proposal.Address.id },
-        });
-        author = profile.profile_name;
-      } catch (e) {
-        author = '';
-      }
-    } else {
-      title = chain ? chain.name : 'Commonwealth';
-      description = '';
-      image = chain
-        ? `https://commonwealth.im${chain.icon_url}`
-        : DEFAULT_COMMONWEALTH_LOGO;
+    let author;
+    try {
+      const profile = await models.Profile.findOne({
+        where: { id: thread.Address.id },
+      });
+      author = profile.profile_name;
+    } catch (e) {
       author = '';
     }
+
+    renderWithMetaTags(res, title, description, author, image);
+  };
+
+  const renderProposal = async (
+    scope: string,
+    res,
+    chain?: ChainInstance
+  ) => {
+    // Retrieve title, description, and author from the database
+    chain = chain || (await models.Chain.findOne({ where: { id: scope } }));
+
+    const title = chain ? chain.name : 'Commonwealth';
+    const description = '';
+    const image = chain
+      ? `https://commonwealth.im${chain.icon_url}`
+      : DEFAULT_COMMONWEALTH_LOGO;
+    const author = '';
+
     renderWithMetaTags(res, title, description, author, image);
   };
 
   app.get('/:scope/proposal/:type/:identifier', async (req, res) => {
     const scope = req.params.scope;
-    const proposalType = req.params.type;
-    const proposalId = req.params.identifier.split('-')[0];
-    await renderProposal(scope, proposalType, proposalId, res);
+    await renderProposal(scope, res);
   });
 
   app.get('/:scope/discussion/:identifier', async (req, res) => {
     const scope = req.params.scope;
-    const proposalType = ProposalType.Thread;
-    const proposalId = req.params.identifier.split('-')[0];
-    await renderProposal(scope, proposalType, proposalId, res);
+    const threadId = req.params.identifier.split('-')[0];
+    if (isNaN(threadId)){
+      return; // don't render because thread ID needs to be a number
+    }
+    await renderThread(scope, threadId, res);
   });
 
   app.get('/:scope/proposal/:identifier', async (req, res) => {
     const scope = req.params.scope;
-    const proposalId = req.params.identifier.split('-')[0];
     const chain = await models.Chain.findOne({ where: { id: scope } });
 
-    // derive proposal type from scope if possible
-    let proposalType;
-    if (chain?.base === ChainBase.CosmosSDK) {
-      proposalType = ProposalType.CosmosProposal;
-    } else if (chain?.network === ChainNetwork.Sputnik) {
-      proposalType = ProposalType.SputnikProposal;
-    } else if (chain?.network === ChainNetwork.Compound) {
-      proposalType = ProposalType.CompoundProposal;
-    } else if (chain?.network === ChainNetwork.Aave) {
-      proposalType = ProposalType.AaveProposal;
-    } else {
+    const proposalTypes = new Set([
+      ChainBase.CosmosSDK,
+      ChainNetwork.Sputnik,
+      ChainNetwork.Compound,
+      ChainNetwork.Aave
+    ]);
+
+    if (!proposalTypes.has(chain?.base)) {
```

Comment:
This logic is not identical to the original logic, which looks at both base and network.

Reply:
Good catch